CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+        -   1.B.2: You are working on a different branch.
+            -   1.B.2.A: You have made commits.
+                -   [Create a fork](#create-a-fork).
+                -   [Add the remote of your fork](#add-a-remote).
+                -   [Push the branch to your fork](#push-a-branch).
+                -   [Clone your fork](#clone-a-repo).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+            -   1.B.2.B: You haven't made commits.
+                -   [Commit your changes](#commit-changes).
+                -   [Create a fork](#create-a-fork).
+                -   [Add the remote of your fork](#add-a-remote).
+                -   [Push the branch to your fork](#push-a-branch).
+                -   [Clone your fork](#clone-a-repo).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+-   2: You have a fork of a Numberscope repository.
+    -   2.A: You haven't made changes.
+        -   [Clone your fork](#clone-a-repo) (if you don't have it on your
+            computer yet).
+        -   [Create a branch](#create-a-branch).
+        -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+    -   2.B: You have made changes.
+        -   2.B.1: You are working on the main branch.
+            -   2.B.1.A: You made commits.
+                -   TODO
+            -   2.B.1.B: You haven't made commits.
+                -   [Stash your changes](#stash-your-changes).
+                -   [Create a branch](#create-a-branch).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+        -   2.B.2: You are working on a different branch.
+            -   2.B.2.A: You have made commits.
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+            -   2.B.2.B: You haven't made commits.
+                -   [Commit your changes](#commit-changes).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+
+## Stash your changes
+
+To "stash" your changes (i.e. squirrel them away for later use), issue the
+following command:
+
+```sh
+git stash
+```
+
+## Create a branch
+
+To create a branch and switch to it in one command, issue the following
+command:
+
+```sh
+git checkout -b your_branch_name_here
+```
+
+At Numberscope we use `snake_case` for branch names.
+
+## Push a branch
+
+To create a branch in a remote Git repository and push your changes to that
+branch, issue the following command:
+
+```sh
+git push -u name_of_remote name_of_branch
+```
+
+The name of the remote is most likely `origin` if you are pushing to a fork of
+one of the Numberscope repositories.
+
+## Create a fork
+
+Note: This is a GitHub operation, not a Git operation.
+
```

Comment:
Maybe add a sentence like "it creates your own private copy of the numberscope repository that is linked to the main repository so that changes can be pushed" or something more accurate than my crude understanding I just demonstrated...